syntax = "proto3";

package payment;

enum Currency{//used lower case for enum members' values because of stripe
  unknown = 0;
  eur = 1;
  usd = 2;
  gbp = 3;
}

message Payment{
  string payment_ref = 1;
  string customer_ref = 2;
  int32 amount = 3;
  Currency currency = 4;
  string card_number = 5;
  int32 card_exp_month = 6;
  int32 card_exp_year = 7;
  string card_cvc = 8;
}

message PaymentRequest{
  Payment payment = 1;
}

message PaymentResponse{
  bool success = 1;
  string message = 2;
}

service  PaymentService{
  rpc pay (PaymentRequest) returns (PaymentResponse);
}